client_test.go

```diff
@@ -3,14 +3,47 @@ package containerd
 import (
 	"context"
 	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
 	"testing"
+	"time"
 )
 
 func init() {
 	flag.StringVar(&address, "address", "/run/containerd/containerd.sock", "The address to the containerd socket for use in the tests")
 	flag.Parse()
 }
 
+const defaultRoot = "/var/lib/containerd-test"
+
+func TestMain(m *testing.M) {
+	// setup a new containerd daemon
+	cmd := exec.Command("containerd", "--root", defaultRoot)
+	// TODO: what todo with IO?
+	if err := cmd.Start(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	time.Sleep(100 * time.Millisecond)
```

Comment:
If you want to block until it is ready, with retries, you can set a WithBackoff and WithTimeout. WithBlock just decides whether it waits on connect or blocks the first call.